### PR TITLE
[Mini Challenge 5] Routing

### PR DESCRIPTION
**Pull request by saulmeneses**
- Updated App routes.
- Updated Header component.
- Updated Auth and Custom Contexts.
- Updated, and added, test cases.
- Added new pages for user favorite videos.
- Applied tests on most components; sharing coverage results (test-coverage.txt, test-capture.png).

### CODE REVIEW

File: src/components/ContentCard/ContentCard.component.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import htmlParser from 'html-react-parser';
+
+import { StyledCard, StyledCardMedia, Title, Description } from './ContentCard.styles';
+
+const ContentCard = ({ item }) => {
+  const history = useHistory();
+  const location = useLocation();
+
+  const isFavoritePage = location.pathname.includes('favs');
+  const path = isFavoritePage ? '/favs/' : '/v/';
```
Comment (jparciga) [inline src/components/ContentCard/ContentCard.component.jsx:13]: 👌🏻

File: src/components/ContentDetails/ContentDetails.component.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    <StyledFlexDiv>
+      <StyledVideoDiv>
+        <StyledIFrame
+          title={item.snippet.title}
+          src={getYoutubeEmbedLink(item.id.videoId)}
+        />
+        <StyledSnippetDiv>
+          <StyledSnippetTitleDiv>
+            <SnippetTitle>{htmlParser(item.snippet.title)}</SnippetTitle>
+            {authenticated && !isFavoritePage && (
```
Comment (jparciga) [inline src/components/ContentDetails/ContentDetails.component.jsx]: I think that it'll be useful to have the "Remove from favorites" button on the favorites page too:

```suggestion
            {authenticated && (
```

File: src/components/LoginDialog/LoginDialog.component.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import { StyledDialog, StyledTextField } from './LoginDialog.styles';
+
+import { useAuth } from '../../providers/Auth';
+
+const LoginDialog = ({ open, onClose }) => {
+  const { login } = useAuth();
+  const [username, setUsername] = useState('wizeline');
+  const [password, setPassword] = useState('Rocks!');
+
+  const handleOnChange = (hook) => ({ target: { value } }) => hook(value);
```
Comment (jparciga) [inline src/components/LoginDialog/LoginDialog.component.jsx:13]: Nice 👌🏻

File: src/components/Header/__tests__/Header.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      <AuthProvider>
+        <WrappedHeader />
+      </AuthProvider>
+    );
+
+    const searchElement = screen.getByLabelText('search');
+    const value = 'wizeline';
+
+    fireEvent.change(searchElement, { target: { value } });
+    expect(searchElement.value).toBe(value);
```
Comment (jparciga) [inline src/components/Header/__tests__/Header.spec.jsx:38]: This is a nice test case

File: src/components/Header/__tests__/Header.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        <WrappedHeader />
+      </AuthContext.Provider>
+    );
+
+    const logOutElement = screen.getByText(/log out/i);
+    expect(logOutElement).toBeInTheDocument();
+
+    fireEvent.click(logOutElement);
+    expect(mockedLogout).toHaveBeenCalled();
+  });
```
Comment (jparciga) [inline src/components/Header/__tests__/Header.spec.jsx:139]: Good test cases 👍🏻

**Review by jparciga — APPROVED** [review]
# Mini-Challenge 5 Feedback
Great job Saul 👏🏻👏🏻🎉!!  Congratulations I think that you really have a great future as a React Engineer 😃!!

Just found a bug: The search bar  is not working when you're on the Favorite video details page:
https://user-images.githubusercontent.com/3632628/113049070-758d1400-9160-11eb-8a0a-fe4ada7cddea.mov

## Acceptance Criteria
✅  All the sections have their own route.
✅  Navigation across the sections is implemented correctly.
✅  The Global State is persistent across all the sections.
✅  The "Mocked Authentication" mechanism works correctly.
✅  The "session data" is stored in the Global Context correctly.
✅  Videos can be added to the Favorites list.
✅  Videos can be removed from the Favorites list.
✅  Navigation to Favorite Videos View using a private route is implemented correctly (only authenticated users should access this section).
✅  Navigation to the Favorite Video Details View using a private route is implemented correctly (only authenticated users should access this section).
✅  Information for the selected favorite video is displayed correctly
✅  The list of other favorite videos in the Favorite Video Details View is displayed.
✅  Testing coverage is above 70%. (Please include a screenshot of the code coverage report in your PR description).

## Bonus Points
[ ] The Add/remove from favorites button should appear when the user passes the mouse over the video card in the list.
[ ] Integrate with a real authentication provider (such as Auth0, OAuth, or Firebase).
✅  A 404 section is shown when a route is not found.
✅  The Login View is implemented as a modal using React Portals.

Screen Shot 2021-03-30 at 13 48 18
Screen Shot 2021-03-30 at 13 48 29

File: src/components/ContentDetails/ContentDetails.component.jsx (hunk 2 above)
Comment (saulmeneses) [reply src/components/ContentDetails/ContentDetails.component.jsx]: Sure thing! I'll commit it

**Pull request closed without merging**